bundles/org.openhab.binding.nuvo/src/main/java/org/openhab/binding/nuvo/internal/communication/NuvoConnector.java

```diff
@@ -0,0 +1,385 @@
+     */
+    public void sendQuery(NuvoEnum zone, NuvoCommand cmd) throws NuvoException {
+        sendCommand(zone.getId() + cmd.getValue() + QUERY);
+    }
+
+    /**
+     * Request the Nuvo controller to execute a command for a zone that takes no arguments (ie power on, power off,
+     * etc.)
+     *
+     * @param zone the zone for which the command is to be run
+     * @param cmd the command to execute
+     *
+     * @throws NuvoException - In case of any problem
+     */
+    public void sendCommand(NuvoEnum zone, NuvoCommand cmd) throws NuvoException {
+        sendCommand(zone.getId() + cmd.getValue());
+    }
+
+    /**
+     * Request the Nuvo controller to execute a command for a zone and pass in a value
+     *
+     * @param zone the zone for which the command is to be run
+     * @param cmd the command to execute
+     * @param value the string value to consider for volume, source, etc.
+     *
+     * @throws NuvoException - In case of any problem
+     */
+    public void sendCommand(NuvoEnum zone, NuvoCommand cmd, @Nullable String value) throws NuvoException {
+        sendCommand(zone.getId() + cmd.getValue() + value);
+    }
+
+    /**
+     * Request the Nuvo controller to execute a configuration command for a zone and pass in a value
+     *
+     * @param zone the zone for which the command is to be run
+     * @param cmd the command to execute
+     * @param value the string value to consider for bass, treble, balance, etc.
+     *
+     * @throws NuvoException - In case of any problem
+     */
+    public void sendCfgCommand(NuvoEnum zone, NuvoCommand cmd, @Nullable String value) throws NuvoException {
+        sendCommand(zone.getConfigId() + cmd.getValue() + value);
+    }
+
+    /**
+     * Request the Nuvo controller to execute a system command the does not specify a zone or value
+     *
+     * @param cmd the command to execute
+     *
+     * @throws NuvoException - In case of any problem
+     */
+    public void sendCommand(NuvoCommand cmd) throws NuvoException {
+        sendCommand(cmd.getValue());
+    }
+
+    /**
+     * Request the Nuvo controller to execute a raw command string
+     *
+     * @param command the command string to run
+     *
+     * @throws NuvoException - In case of any problem
+     */
+    public void sendCommand(String command) throws NuvoException {
+        String messageStr = BEGIN_CMD + command + END_CMD;
+
+        logger.debug("sending command: {}", messageStr);
+
+        OutputStream dataOut = this.dataOut;
+        if (dataOut == null) {
+            throw new NuvoException("Send command \"" + messageStr + "\" failed: output stream is null");
+        }
+        try {
+            // Essentia G needs time to wake up when in standby mode
+            // I don't want to track that in the binding, so just do this always
+            if (this.isEssentia) {
+                dataOut.write(WAKE_STR);
+                dataOut.flush();
+                Thread.sleep(5);
```

Comment:
This device's spec document explicitly mention 5ms for sleep wakeup and 50 milliseconds between commands. It this going to be a problem with the thread pools?

Reply:
No, that won't be a problem. But values shorter than 100ms can be ignored by the JVM completely. To increase the probability for the JVM to respect the sleeps, you should set them to 100ms.

Reply:
Would it be better to do something with wait() instead of Thread.sleep()?

Reply:
No, that won't make things better. I would try if the sleeps are necessary at all. 5ms might be the overall system-to-system latency. Otherwise, it's ok to go with the 100ms delays.

Reply:
I removed it and the device still wakes up properly.